### ISSUE TITLE
[Bug]: UX issues in MultiSelect widget

### ISSUE THREAD

**Issue by shadabbuchh** [appsmithorg/appsmith#22984]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

1. Using Keyboard nav we cannot select the `Select All` option:

https://user-images.githubusercontent.com/39921438/236135032-03fb66cd-f857-45a7-b62e-3244cec480cc.mov

2. The cursor shifts to the beginning of the search text once we navigate the options using a keyboard:

https://user-images.githubusercontent.com/39921438/236135414-8e6bf83e-4655-4329-aa0c-8bfdfb3c066b.mov

### Steps To Reproduce

1. D&D a MultiSelect widget.
2. Open the options dropdown & try to navigate using keyboard.

### Environment

Production

### Version

Cloud
